src/createAnimatedComponent/InlinePropManager.ts

```diff
@@ -35,14 +45,20 @@ function inlinePropsHasChanged(
   return false;
 }
 
-function getInlinePropsUpdate(inlineProps: Record<string, any>) {
+function getInlinePropsUpdate(
+  inlineProps: Record<string, unknown> &
+    TransformsStyle &
+    Partial<TransformArrayItem>
+) {
   'worklet';
-  const update: Record<string, any> = {};
+  const update: Record<string, unknown> = {};
   for (const [key, styleValue] of Object.entries(inlineProps)) {
     if (key === 'transform') {
-      update[key] = styleValue.map((transform: Record<string, any>) => {
-        return getInlinePropsUpdate(transform);
-      });
+      update[key] = inlineProps[key]?.map(
```

Comment:
This change potentially changes the behavior. Please try not to change how is JS executed on type fixing pull request. Also, why we stop using `styleValue` here?

Reply:
We have `const [key, styleValue] of Object.entries(inlineProps)` so `styleValue` is equal to `inlineProps[key]`. However due to typescript limitations inside conditional
`if (key === 'transform')` only type of `key` is narrowed.

Reply:
Previously we were casting `styleValue` to `any` to use it like an array. But when we substitute it with `inlineProps[key]` then it is automatically typed as `TransformsStyle`, so we TS knows it is indeed an array.

Reply:
I meant the `?` operator. Before we just called `map`, so calling improperly would throw an error. Now we check if it's available, altering the behavior.